### ISSUE TITLE
hands-on evaluation of new AAD policy to protect device join - phase 1

### ISSUE THREAD

**Issue by tkol2022** [cisagov/ScubaGear#291]
# 💡 Summary #

We received a recommendation from Microsoft to create a new baseline policy to protect the MFA registration process via a conditional access policy. Also references were made by Microsoft's Detection and Response Team and Mandiant that align with Microsoft's formal comment on the baseline. That was explored in #273.

This issue explores a related action, device join, which centers on a user "joining" a laptop or other device to Azure AD, effectively making the machine "managed". We explored the question, **does it make sense to create a new baseline policy that secures the joining of devices to AAD? What problems does this solve? What constraints are there?**

Hands-on prototyping was performed. Details are provided below.

**Comment by tkol2022** [comment]
# Prototyping Details

## Test Scenario - New join device CA policy requiring MFA
	Include specific users (for test)
	Cloud apps or actions > User actions > Register or join devices
	Grant > Grant access > Require multifactor authentication

To implement a device join policy in conditional access I set the User Actions value to "Register or join devices".

![image](https://user-images.githubusercontent.com/107076927/235011874-0a4f7765-a473-474f-857d-8298c668e937.png)

## Test Results
Learned some interesting info from testing this. First when you scope the CA policy to "Register or join devices", you can only select Grant > Grant access > Require multifactor authentication. You cannot choose to Block access so I was unable to create a policy that requires the user to be on a trusted network in order to join the device (like when testing MFA registration scenarios). Also you cannot select other options in the Grant access list such as Require a compliant device - this is likely because Microsoft wants to avoid the "chicken before the egg".

I also learned that, without this new CA policy, I was able to register a new Windows machine with Azure AD without needing to provide MFA credentials. Once I created the new CA policy and repeated the process to register a fresh machine, only then did the Windows machine prompt me to provide MFA credentials. This was consistent on all test tenants (G5, E5, G3). This is specific to "registering" a device with AAD - "joining" a device is a different action - see below for that.

On the G5 and E5 tenants, "joining" a device to AAD seemed to require MFA by default, even though I didn't specifically configure this to be required in the tenant Devices > Device Settings or in a new CA policy. On those tenants by default registering a device did not require MFA.

On the G3 tenant however, by default, registering or joining a device to AAD did not require MFA. Once I created a new CA policy requiring MFA to perform registering or joining, the policy worked as expected and I started receiving the MFA prompt.

On the GCCHigh tenant, by default, registering or joining a device to AAD did not require MFA. Interestingly in the GCCHigh tenant there is no option to select "Register or join devices" in AAD conditional access, so that feature seems to be missing. Instead, for GCCHigh, I used an alternative setting AAD Devices > Device Settings and set the toggle box named "Require Multi-Factor Authentication to register or join devices with Azure AD" to Yes.

![image](https://user-images.githubusercontent.com/107076927/235012643-759a8cb1-d96c-4db7-9d0b-f0662d0ab64f.png)

## Summary and Recommendation
In summary, since joining a device to AAD by default can be done without MFA in some tenants, I think it is worth creating a new policy to protect the device join process. Since the conditional access feature "Register or join devices" is missing in the GCCHigh tenant, it might be better to base the policy implementation on the Devices > Device Settings feature instead.

If we end up implementing this policy, the respective Powershell cmdlet to retrieve the Device Settings data from AAD is "Get-MgPolicyDeviceRegistrationPolicy" and the specific field to check is named "MultiFactorAuthConfiguration". A value of 1 in that field means that the tenant is configured to Yes, require MFA to join devices and a value of 0 means the opposite.

![image](https://user-images.githubusercontent.com/107076927/235012796-f7eb76c0-d128-4bec-8dea-ff2627f85792.png)

**Comment by tkol2022** [comment]
## Question to CISA
Based on testing to understand the feasibility and constraints of the proposed policy, I recommend a new baseline policy to protect device join. See previous comment for details.

- Does CISA agree with this recommendation?
- If yes, any specific direction on the policy?

**Comment by tkol2022** [comment]
Since performing the hands-on evaluation for this issue, I reviewed a related comment from the feedback spreadsheet which proposed an additional technical policy to further strengthen the device join process. This would be in addition to the policy that I already tested and proposed in this issue.
We do not have the time to evaluate this additional policy during Emerald so I created a separate backlog issue #314 and labelled that as Phase 2. Once Phase 2 has completed we will have a better idea on what the new baseline policy or policies (if multiple are needed) will look like and their operational impacts based on hands-on testing.
